### PR TITLE
[SPARK-6212][SQL]The EXPLAIN output of CTAS only shows the analyzed plan

### PR DESCRIPTION
**Pull request by yjshen**
JIRA: https://issues.apache.org/jira/browse/SPARK-6212

### CODE REVIEW

**Comment by yjshen** [comment]
The reason for the bug:

Give `EXPLAIN CREATE TABLE t1 SELECT * FROM exist_table` as an example.
`ExplainCommand(CreateTableAsSelect)` is used internally to run the command:
```scala
val queryExecution = sqlContext.executePlan(logicalPlan/*It is CreateTableAsSelect in this case*/)
val outputString = if (extended) queryExecution.toString else queryExecution.simpleString
```
After analysing phase, createTableAsSelect was converted to another RunnableCommand, which is a leaf operator by default:
``` scala
private[sql] trait RunnableCommand extends LogicalPlan with logical.Command {
  override def children: Seq[LogicalPlan] = Seq.empty
```
Therefore, no further optimisation or planning is feasible in this case.

**Comment by yjshen** [comment]
Currently, I just implement `def children: Seq[LogicalPlan]` in CreateTableAsSelect to make it work, but I'm not sure if other reasons exist to make `RunnableCommand` have no children.
If the current implementation is considered acceptable, other `RunnableCommand`s has logical plan should also do the same.
@rxin @yhuai

**Comment by chenghao-intel** [comment]
duplicated with #7980 :)

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/CreateTableAsSelect.scala [hunk 1]
```diff
@@ -91,6 +93,6 @@ case class CreateTableAsSelect(
   }
 
   override def argString: String = {
-    s"[Database:$database, TableName: $tableName, InsertIntoHiveTable]\n" + query.toString
+    s"[Database:$database, TableName: $tableName, InsertIntoHiveTable]"
```
Comment (yhuai) [inline sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/CreateTableAsSelect.scala]: @chenghao-intel I feel this string is better than having a string containing `tableDesc` (I feel it has way too much information for a user).

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveExplainSuite.scala [hunk 2]
```diff
@@ -74,4 +74,21 @@ class HiveExplainSuite extends QueryTest {
...
+         |AS
+         |SELECT * FROM src
+      """.stripMargin), true,
+      "== Parsed Logical Plan ==",
+      "== Analyzed Logical Plan ==",
+      "== Optimized Logical Plan ==",
+      "== Physical Plan ==",
+      "CreateTableAsSelect",
+      "InsertIntoHiveTable",
+      "src")
```
Comment (yhuai) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveExplainSuite.scala]: I feel this one is pretty much the same as the test in `explain create table command` (the above test). I think we need to have a test to make sure that the physical plan output string does not contain `Subquery`.

**Comment by yhuai** [comment]
@chenghao-intel How about we help @yjshen finish this one? What do you think?

@yjshen can you update the test?

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/CreateTableAsSelect.scala (hunk 1 above)
Comment (chenghao-intel) [inline sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/CreateTableAsSelect.scala]: But `tableDesc` will be helpful if user want to get more information on what backend for the created table. 
Like will it be parquet format by default in spark sql, which is not the same as hive does.

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveExplainSuite.scala [hunk 3]
```diff
@@ -74,4 +79,30 @@ class HiveExplainSuite extends QueryTest {
...
+      read.json(rdd).registerTempTable("jt")
+      val outputs = sql(
+        s"""
+           |EXPLAIN EXTENDED
+           |CREATE TABLE t1
+           |AS
+           |SELECT * FROM jt
+      """.stripMargin).collect().map(_.mkString).mkString
+
+      val shouldContain =
```
Comment (chenghao-intel) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveExplainSuite.scala]: Can you also to check if it's real the optimized logical plan?
https://github.com/apache/spark/pull/7980/files#diff-dc57b297e91eedbf8cd67806640a6bbcR75

Free feel to take it. :)

Comment (yjshen) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveExplainSuite.scala]: EliminateSubQueries also take effect during optimiser, isn't it?

Comment (yhuai) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveExplainSuite.scala]: `EliminateSubQueries` is the batch in the optimizer.

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/CreateTableAsSelect.scala (hunk 1 above)
Comment (yhuai) [inline sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/CreateTableAsSelect.scala]: I think it is fine to add the storage format in the string. But, `tableDesc` itself has too much information and it is hard to figure out the meaning.

**Comment by yhuai** [comment]
LGTM. I am merging it to master and branch 1.5.

**Pull request closed without merging**